### PR TITLE
Add higher tiered fluid regulators

### PR DESCRIPTION
**Pull request by kstvr32**
With changes from @Glease (GTNewHorizons/GT5-Unofficial#890), fluid regulators are no longer limited to storing their data in a single `int`. This makes it easy to add higher tiered fluid regulators to match the pumps.

This can be particularly useful for XL steam turbines in later tiers.

## Adds `Fluid Regulator (UHV)` and `Fluid Regulator (UEV)` as usable covers

Screenshot 2024-09-07 182736

### UHV
uhv
uhv_crafting

### UEV
uev
uev_crafting

## Adds `Fluid Regulator (UIV)`, `Fluid Regulator (UMV)`, `Fluid Regulator (UXV)` and, `Fluid Regulator (MAX)` as crafting components

These are not usable covers matching the behavior of other high tiered pumps/conveyers. Instead, they can be used as crafting components.

top_tiers
(UMV, UXV, and MAX are animated)

uiv_crafting
umv_crafting
uxv_crafting
MAX does not have a crafting recipe as there are no craftable MAX circuits or MAX pumps.

### CODE REVIEW

File: src/main/java/gregtech/api/enums/ItemList.java [hunk 1]
```diff
@@ -445,6 +445,8 @@ public enum ItemList implements IItemContainer {
     FluidRegulator_LuV,
     FluidRegulator_ZPM,
     FluidRegulator_UV,
+    FluidRegulator_UHV,
+    FluidRegulator_UEV,
```
Comment (OneEyeMaker) [inline src/main/java/gregtech/api/enums/ItemList.java:449]: Why don't you add all tiers (up to and including MAX)? At least for the sake of consistency with other components. Or is there some limitation or balance concern?

Comment (kstvr32) [reply src/main/java/gregtech/api/enums/ItemList.java:449]: Pumps pants UEV are just crafting components. They can’t be used as covers. 

I could add all of the regulators, leave them uncraftable and unusable.

Comment (kstvr32) [reply src/main/java/gregtech/api/enums/ItemList.java:449]: Thinking about it more, we could add even higher tiered pumps and regulators due to the change with how data is stored. Not sure if there are balance concerns.

Comment (OneEyeMaker) [reply src/main/java/gregtech/api/enums/ItemList.java:449]: I think there should be regulators past UEV, at least as crafting components. If there are no balance concerns, then both pump and regulator covers can be registered for those tiers.

**Review by chochem — CHANGES_REQUESTED** [review]
this pr appears to be broken and crashes the nightly https://discord.com/channels/181078474394566657/603348502637969419/1281823994806276187

please test your changes next time

File: src/main/java/gregtech/common/items/IDMetaItem01.java [hunk 2]
```diff
@@ -423,7 +423,9 @@ public enum IDMetaItem01 {
     Tool_Scanner(762),
     NC_SensorKit(763),
     Duct_Tape(764),
-    McGuffium_239(765);
+    McGuffium_239(765),
+    FluidRegulator_UHV(766),
+    FluidRegulator_UEV(767);
```
Comment (chochem) [inline src/main/java/gregtech/common/items/IDMetaItem01.java:428]: yea those indices are definetly out of bounds

Comment (kstvr32) [reply src/main/java/gregtech/common/items/IDMetaItem01.java:428]: Can I move this to metaitem 2/3?

**Comment by kstvr32** [comment]
> this pr appears to be broken and crashes the nightly https://discord.com/channels/181078474394566657/603348502637969419/1281823994806276187
> 
> please test your changes next time

This ran fine on my local. I can check again this afternoon. 

I thought there was enough room given that the signed short can be 32,767 giving 767 non-generated items.

**Comment by chochem** [comment]
> This ran fine on my local.

pretty sure thats impossible

**Comment by boubou19** [comment]
it's possible it worked in dev env, but not fullpack

**Comment by chochem** [comment]
> it's possible it worked in dev env, but not fullpack

it wouldnt crash but the items still wouldnt be there

**Comment by kstvr32** [comment]
Sorry about that, I’ll fix this and implement @OneEyeMaker suggested changes. 

Do you have an recommendations on a free set of IDs I can move this to?

**Comment by Dream-Master** [comment]
@kstvr32 please finish the pr add the missing tiers and test your pr first. After this you can undraft it and ask for a re review.

File: src/main/java/gregtech/common/items/IDMetaItem01.java (hunk 2 above)
Comment (kstvr32) [reply src/main/java/gregtech/common/items/IDMetaItem01.java:428]: Moving to a free range near the beginning of item 1.

**Comment by kstvr32** [comment]
Fixed PR. Implemented UHV and UEV fluid regulators correctly.

Still need texture for UMV, UXV, and MAX.

@OneEyeMaker should I add recipes for UIV, UMV, UXV, and MAX if they are not used as covers and not as crafting components (yet)?

**Comment by Dream-Master** [comment]
Sure add textures and recipes @kstvr32

**Comment by kstvr32** [comment]
Added textures and recipes, re-opening PR.

File: src/main/java/gregtech/api/enums/ItemList.java (hunk 1 above)
Comment (kstvr32) [reply src/main/java/gregtech/api/enums/ItemList.java:449]: Added fluid regulators all the way to MAX tier.

**Review by boubou19 — APPROVED** [review]

**Pull request merged**